3scaleAdapter/httpPluginClient/httpPluginClient_test.go

```diff
@@ -0,0 +1,120 @@
+package httpPluginClient
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+)
+
+const (
+	validUserKey     = "12345"
+	validAccessToken = "abcdef"
+	validServiceID   = "111111"
+)
+
+func Test_Authorize_Ok(t *testing.T) {
+
+	systemURL, _ := url.Parse("https://mysystemurl-admin.3scale.net/")
+
+	originalRequest := &http.Request{
+		Method: "get",
+		URL: &url.URL{
+			User: &url.Userinfo{},
+			Path: "/path?query=1&query=2&query=3&user_key=" + validUserKey,
+		},
+		Header: map[string][]string{"user-key": {"user_key_header"}},
+	}
+
+	c := NewClient(nil)
+
+	ts := HTTPPluginMock(t)
+
+	c.BaseURL, _ = url.Parse(ts.URL)
+
+	ok, err := c.Authorize(validAccessToken, validServiceID, systemURL, originalRequest)
+	if err != nil {
+		t.Errorf("Authorize returned error: %#v", err)
+	}
+
+	if ok != true {
+		t.Error("Expected true")
+	}
+}
+
+func Test_Authorize_Failing(t *testing.T) {
+
+	systemURL, _ := url.Parse("https://mysystemurl-admin.3scale.net/")
+
+	originalRequest := &http.Request{
+		Method: "get",
+		URL: &url.URL{
+			User: &url.Userinfo{},
+			Path: "/path?query=1&query=2&query=3&user_key=" + validUserKey,
+		},
+		Header: map[string][]string{"user-key": {"user_key_header"}},
+	}
+
+	c := NewClient(nil)
+
+	ts := HTTPPluginMock(t)
+
+	c.BaseURL, _ = url.Parse(ts.URL)
+
+	ok, err := c.Authorize("aaaaaa", validServiceID, systemURL, originalRequest)
+	if err != nil {
+		t.Errorf("Authorize returned error: %#v", err)
+	}
+
+	if ok != false {
+		t.Error("Expected false")
+	}
```

Comment:
These two tests look fine but I would probably move them to single test `TestAuthorize` and vary the input and expected output with [table driven tests](https://github.com/golang/go/wiki/TableDrivenTests). Thats just my personal preferance though and its common pattern in Go codebases, but feel free to ignore.

Reply:
Nope, that's a good point. I will review that, thanks!